Bottom: f79eba2aa01d069f8f10bbfe30fe7fc28360448f
Top:    eb4d566b8f4adf36af4c24d26c6725ab8a656462
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-05 18:51:35 +0800

Refresh of autofs-5.0.7-fix-memory-leak-in-get_dc_list.patch

---

diff --git a/modules/dclist.c b/modules/dclist.c
index d16b913..bd4d5e6 100644
--- a/modules/dclist.c
+++ b/modules/dclist.c
@@ -549,6 +549,8 @@ struct dclist *get_dc_list(unsigned int logopt, const char *uri)
 					error(logopt,
 					      "invalid port: %u", dcs[i].port);
 					free_srv_rrs(dcs, numdcs);
+					/* Ensure realloced list is freed */
+					list = tmp;
 					goto out_error;
 				}
 				strcat(tmp, port);
